components/ILIAS/OpenIdConnect/classes/class.ilOpenIdConnectSettingsGUI.php

```diff
@@ -498,68 +438,231 @@ protected function prepareRoleSelection(bool $a_with_select_option = true): arra
     protected function profile(ilPropertyFormGUI $form = null): void
     {
         $this->checkAccess('read');
-        $this->setSubTabs(self::STAB_PROFILE);
-
-        if (!$form instanceof ilPropertyFormGUI) {
-            $form = $this->initProfileForm();
-        }
-        $this->mainTemplate->setContent($form->getHTML());
+        $this->chooseMapping();
+        $this->userMapping();
     }
 
-    protected function initProfileForm(): ilPropertyFormGUI
+    protected function scopes(ilPropertyFormGUI $form = null): void
     {
+        $this->checkAccess('read');
+        $this->setSubTabs(self::STAB_SCOPES);
+        $form = $this->initScopesForm();
+        $this->tpl->setContent($this->renderer->render($form));
+    }
+    protected function initScopesForm(ilPropertyFormGUI $form = null)
+    {
+        $this->checkAccess('read');
+
         $form = new ilPropertyFormGUI();
-        $form->setTitle($this->lng->txt('auth_oidc_mapping_table'));
-        $form->setFormAction($this->ctrl->getFormAction($this, 'saveProfile'));
+        $ui_container = [];
+        $ui_container = $this->buildScopeSelection($ui_container);
+
+        $form = $this->ui->input()->container()->form()->standard($this->ctrl->getFormAction($this, 'saveScopes'), $ui_container);
+
+        return $form;
+    }
+
+    protected function buildScopeSelection(array $ui_container): array {
```

Comment:
Please try to provide the shape of the arrays according to: https://phpstan.org/writing-php-code/phpdoc-types

And:
```suggestion
    private function buildScopeSelection(array $ui_container): array {
```